### ISSUE TITLE
make test fails with Python 3.

### ISSUE THREAD

**Issue by BreD1810** [herd/herdtools7#149]
My machine uses python 3.9.1, which causes this error with `make test`:
```
sh ./dune-build.sh $HOME
+ ./version-gen.sh /home/brad
+ dune build --profile release @all
Done: 4798/4800 (jobs: 1)
dune runtest --profile=release
  shelf_test alias internal/lib/tests/runtest (exit 2)
(cd _build/default/internal/lib/tests && ./shelf_test.exe)
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
NameError: name 'execfile' is not defined
Failed Shelf.of_file reads simple valid files: raised exception
Fatal error: exception Command.Error("Process returned error code 1")
Done: 1138/1141 (jobs: 1)make: *** [Makefile:78: dune-test] Error 1
```

It also fails in our CI running Python 2.7.

Python 3.9.1 can be supported by changing the `execfile` call (line 66 `internal/lib/shelf.ml`) to:
```ocaml
    Printf.fprintf chan "exec(open(%s).read(), m)\n" (Filename.quote (Filename.basename path)) ;
```
Note that `execfile` was [removed in Python 3.3](https://docs.python.org/3.3/whatsnew/3.0.html?highlight=execfile#builtins) so I'm not sure which Python version this is targetting?
Perhaps the targeted Python version should be documented in `install.md`?

**Comment by eth-arm** [comment]
What OS & version is your machine & your CI running?

**Comment by BreD1810** [comment]
My machine is running Python 2.7.16, OS is the latest version of Arch Linux.

The CI running in a docker container, which you can run locally by pulling `registry.gitlab.com/bred1810/docker-herdtools:latest`. Dockerfile is [here](https://gitlab.com/BreD1810/docker-herdtools/-/blob/master/Dockerfile).
This uses Python 2.7.16, OS is Debian 10.

**Comment by eth-arm** [comment]
PR #152 adds support for both systems where `python` is Python 2 and where it is Python 3.

However, `execfile` [definitely exists in Python 2.7](https://docs.python.org/2.7/library/functions.html#execfile), so that should only fail if the Python binary that is actually being run is not actually 2.7.

**Comment by BreD1810** [comment]
Ah, you're right. I misremembered the CI's error when investigating on my machine. That error appears related to #150.
PR #152 fixes this issue for me. Thank you for fixing that.

**Comment by BreD1810** [comment]
Resolved by #152
